### PR TITLE
Add App Configuration to environment

### PR DESCRIPTION
**Pull request by TZdybel**
Needed for: https://github.com/hashicorp/terraform-provider-azurerm/pull/23789

### CODE REVIEW

File: sdk/environments/helpers.go [hunk 1]
```diff
@@ -25,6 +25,16 @@ func ApiManagementAPI(domainSuffix string) Api {
 	}
 }
 
+func AppConfigurationAPI(domainSuffix string) Api {
+	return &ApiEndpoint{
+		domainSuffix:       pointer.To(domainSuffix),
+		endpoint:           nil,
+		appId:              pointer.To(apiManagementAppId),
+		name:               "ApiManagement",
```
Comment (tombuildsstuff) [inline sdk/environments/helpers.go:33]: can we update these to be:

```suggestion
		appId:              pointer.To(appConfigurationAppId),
		name:               "AzureAppConfiguration",
```

**Review by tombuildsstuff — APPROVED** [review]
hey @TZdybel 

Thanks for this PR - I've taken a look through and left one comment inline, but if we can fix that up then this otherwise LGTM 👍

File: sdk/environments/helpers.go (hunk 1 above)
Comment (TZdybel) [reply sdk/environments/helpers.go]: Sorry for that, should have seen that earlier.
Already fixed :)

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**